File: src/leetcode/python/StudentAttendanceRecordI_551.py
# def checkRecord(self, s):
#     return not (s.count('A') > 1 or 'LLL' in s)


class StudentAttendanceRecordI_551(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        numA, numL = 0, 0
        conL = False
        for c in s:
            if c == 'A':
                numA += 1
            elif c == 'L':
                numL += 1
                if numL == 3:
                    conL = True
                continue
            numL = 0
        return not (numA > 1 or conL)

File: src/leetcode/java/StudentAttendanceRecordI_551.java
package leetcode.java;

public class StudentAttendanceRecordI_551 {
  public boolean checkRecord(String s) {
    char[] chs = s.toCharArray();
    int numA = 0;
    int numL = 0;
    boolean contThreeL = false;
    for (char c : chs) {
      if (c == 'A') {
        numA++;
      } else if (c == 'L') {
        numL++;
        if (numL == 3) contThreeL = true;
        continue;
      }
      numL = 0;
    }
    return !(numA > 1 || contThreeL);
  }
}
